File: 22-07-2022/es1.py
import os
import sys


def printFileSystem(path, level):
	files = os.listdir(path)
	for fl in files:
		if os.path.isdir(os.path.join(path, fl)):
			printFile(fl, level)
			printFileSystem(os.path.join(path, fl), level+1)
		else:
			printFile(fl, level)
		
def printFile(fl, level):
	print("\t"*level, fl)

if __name__ == "__main__":
	path = os.sys.argv[1]
	printFileSystem(path, 0)

	
	
	

File: 22-07-2022/es1.c
#include <sys/types.h>
#include <dirent.h>
#include <dirent.h>
#include <stdio.h>
#include <libgen.h>
#include <string.h>
#define MAXPATH 1024

void printFile(char *s, int lvl){
	for(int i=0; i<lvl; i++){
		printf("\t");
	}
	printf("%s\n", s);
}

void printTree(char* path, int lvl) {
	DIR* dir = opendir(path);
	if (dir == NULL) {
		printf("cannot open %s\n", path);
		return;
	}
	
	struct dirent *obj;
	while( (obj = readdir(dir)) != NULL) {
		if( strncmp(obj->d_name, ".", 1) == 0) {
			continue;
		}
		printFile(obj->d_name, lvl);
		
		if(obj->d_type == DT_DIR) {
			char buffer[MAXPATH];
			snprintf(buffer, MAXPATH, "%s/%s", path, obj->d_name);
			printTree(buffer, lvl+1);		
		}	
	}
}

int main(int argc, char* argv[]) {
	
	if(argc != 2) {
		perror("too many arguments");
	}
	
	printTree(argv[1], 0);
	return 0;
}
